TTHAnalysis/python/tools/higgsRecoTTH.py

```diff
@@ -29,52 +33,164 @@ def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
     def analyze(self, event):
         writeOutput(self, self.run(event, NanoAODCollection))
         return True
-
     # code
     def run(self,event,Collection):
+## TODO ##        
+    # status flag for gen particles
+    # ----------------------------- 
+        statusFlagsMap={
+        'isHardProcess' : 7,
+        'isPrompt'      : 0
+        }
+    # define variables and gen collections
+    # ------------------------------------   
+        #HiggsDaughters = genHiggsDaughtersSelection(genpar) # that is how you define a collection from genproducer, i.e. apply the selection on your collection and it return a filtered collection
+        genjet = Collection(event,"GenJet","nGenJet")
+        genpar = Collection(event,"GenPart","nGenPart")
+        QFromWFromH = []
+        LFromWFromH = []
+        QFromWFromT = []
+        LFromWFromT = []
+        #gengoodJets = []
+        matchedpartons          = 0
+        bothmatchedpartons      = 0 
+        mismatchedtoptaggedjets = 0
+        pTHgen = 0
+        delR_H_partons = 0
+        delR_H_j1j2    = 0
         
-        nleps = getattr(event,"nLepGood")
-        nFO = getattr(event,"nLepFO"+self.label)
-        ileps = getattr(event,"iLepFO"+self.label)
-        leps = Collection(event,"LepGood","nLepGood")
-        lepsFO = [leps[ileps[i]] for i in xrange(nFO)]
-        jets = [x for x in Collection(event,"JetSel"+self.label,"nJetSel"+self.label)]
-        ret = {}        
+
+    #def isAncestor(a,p): #TODO need to modify this function for my purpose fitting the loop below
```

Comment:
If the function compiles, then please leave it uncommented (but unused) until you fix it.

Reply:
Note this, perhaps still todo?